CHI_TECH/ChiMesh/Cell/cell.h

```diff
@@ -4,6 +4,25 @@
 #include"../chi_mesh.h"
 #include <tuple>
 
+namespace chi_mesh
+{
+  enum CellTypes
+  {
+    GHOST_CELL = 0,
+    SLAB_CELL = 1,
+    SPHERICAL_SHELL_CELL = 2,
+    CYLINDRICAL_ANNULUS_CELL = 3,
+    TRIANGLE_CELL = 4,
+    QUADRILATERAL_CELL = 5,
+    POLYGON_CELL = 6,
+    TETRAHEDRON_CELL = 7,
+    HEXAHEDRON_CELL = 8,
+    POLYHEDRON_CELL = 9
+  };
+}
```

Comment:
Unless you have a strong objection I would like to not do this. Here's why:
- Functionally it doesn't make a difference (unless you can correct me)
- The CellTypes enum is defined in this file because it seems to be a logical place to have it. The manner in which we add it to the namespace reads as "append this to namespace chi_mesh". I would like to keep this notion. It reads well.
- Adding an indentation level here then opens us to question of why we don't encapsulate all the derived cell types also. Reason, cause its not necessary.

As I said. If you have a strong technical argument against then I will do it like you suggest, otherwise I like how this reads.

Reply:
I agree with and don't have anything against adding the `CellTypes` enum to `namespace chi_mesh` in this file. What I didn't like is the definition of the class `Cell`, which is part of the namespace too, outside of that namespace. I personally like if everything that relates to the namespace symbols to be 'done in the namespace'. If I wanted to search for all places where classes and functions in that namespace are *defined*, I can then just search for  `namespace chi_mesh {`; if I wanted to know where objects of that namespace are being used, I'd search for `chi_mesh::`. I also think that this

```c++
namespace chi_mesh {

class CellSlab : public Cell
{
  ...
};

} // namespace chi_mesh
```

is more readable than

```c++
class chi_mesh::CellSlab : public chi_mesh::Cell
{
   ...
}
```

But I am not too strongly advocating for it – @wdhawkins, what's your opinion about that?

As for the indentation, I don't mind at all not indenting the definitions in the namespace, as seen in the snippet above.

Reply:
I think my preferred style would be:

`namespace chi_mesh {

class CellSlab : public Cell
{`